adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -302,7 +305,10 @@ boolean writeFileToLocation(CreateStorageRequest createStorageRequest, String lo
           }
         };
 
-    return sendHttpRequest(httpPut, responseHandler);
+    boolean successful = sendHttpRequest(httpPut, responseHandler);
```

Comment:
Can you declare this boolean above the try{}
Then put this in the try{}: successful = sendHttpRequest(httpPut, responseHandler)
So that the tempFile.delete() can happen in a finally{}?

I see that you do deleteOnExit but I think that only carries when the program or JVM closes, but this is meant to be a long running process so we could, in an error state, and up with a lot of temp files that never got deleted.

As it stands now if there is an IOException in the try, after the temp files is created, the temp file wouldn't be deleted until the JVM shuts down.